flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/ManifestOutputFileFactory.java

```diff
@@ -41,6 +42,19 @@ class ManifestOutputFileFactory {
   private final long attemptNumber;
   private final AtomicInteger fileCount = new AtomicInteger(0);
 
+  public ManifestOutputFileFactory(
+      Supplier<Table> tableSupplier,
+      Map<String, String> props,
+      String flinkJobId,
+      String operatorId) {
+    this.tableSupplier = tableSupplier;
+    this.props = props;
+    this.flinkJobId = flinkJobId;
+    this.operatorUniqueId = operatorId;
+    this.subTaskId = 0;
```

Comment:
does this constructor make sense? is this correct? note that subTaskId and attemptNumber were used in file name patterns in this class. 

```
  private String generatePath(long checkpointId) {
    return FileFormat.AVRO.addExtension(
        String.format(
            "%s-%s-%05d-%d-%d-%05d",
            flinkJobId,
            operatorUniqueId,
            subTaskId,
            attemptNumber,
            checkpointId,
            fileCount.incrementAndGet()));
```

On paper, setting them to 0 can cause collisions and overrides. Granted that commit always happen on subtask 0 with the `global` shuffle. Hence subTaskId may not matter much. But `attemptId` could still cause override behavior.

Reply:
I think we can use `getRuntimeContext().getAttemptNumber()` from the `IcebergWriteAggregator.open` to get the required parameters, and then we don't need to use hardcoded values.

Reply:
I forgot to actually remove the aforementioned constructor. So now we only have 1 constructor